lib/active_merchant/billing/gateways/sage_pay.rb

```diff
@@ -37,13 +37,20 @@ class SagePayGateway < Gateway
         :jcb => "JCB"
       }
 
-      AVS_CVV_CODE = {
+      AVS_CODE = {
         "NOTPROVIDED" => nil,
         "NOTCHECKED" => 'X',
         "MATCHED" => 'Y',
         "NOTMATCHED" => 'N'
       }
 
+      CVV_CODE = {
+        "NOTPROVIDED" => 'S',
+        "NOTCHECKED" => 'X',
+        "MATCHED" => 'M',
```

Comment:
Looks good. I wonder if we have any other downstream dependencies on the 'Y' value other than risk_analysis. Definitely worth a look